### DISCUSSION TITLE
SIP-12 (Domain Resolution)

### DISCUSSION THREAD

**Discussion by hmalik88** [MetaMask/SIPs#103] [General]
This thread is intended for discussion regarding the addition of a name resolution API.

**Comment by pawmmm** [comment]
@hmalik88 @matthewwalsh0 As [currently implemented](https://github.com/MetaMask/snaps/pull/1768) if the domain does not match the ENS character standard as defined in [export function isValidDomainName(address)](https://github.com/MetaMask/metamask-extension/blob/c40dbb1ec72b91827debb2fffba4097a838f21c7/ui/helpers/utils/util.js#L109) the Extension shows INVALID_RECIPIENT_ADDRESS_ERROR instead, even if OnNameLookupHandler resolves to valid public address.

Different naming systems may support different character sets and formats. For example [FIO Protocol](https://fio.net/) follows something@something format, which would not be supported by existing implementation:

![image](https://github.com/MetaMask/SIPs/assets/6402117/0619c394-66d4-44f2-ad9b-8d03e997c879)

![image](https://github.com/MetaMask/SIPs/assets/6402117/7f39e38f-3969-4d77-8e85-701dff80f555)

Recommendation:
This can be corrected by implementing one of below:
1. Remove [isValidDomainName](https://github.com/MetaMask/metamask-extension/blob/c40dbb1ec72b91827debb2fffba4097a838f21c7/ui/helpers/utils/util.js#L109) altogether and move ENS domain validation to the ENS resolver.
2. Suppress INVALID_RECIPIENT_ADDRESS_ERROR if OnNameLookupHandler returns valid public address.

**Reply by hmalik88** [reply]
Hi Pawel,

Removing `isValidDomainName` altogether is not a route we can take. Doing so would mean a lot of unnecessary computation in the extension. Instead, we are allowing for snaps to state their TLDs and thinking of allowing for stating of non-TLD domains to be referenced through some prefix scheme in the input e.g. `fio:pawel@edge`.

**Reply by pawmmm** [reply]
@hmalik88 Thanks. That's fine, the problem is that INVALID_RECIPIENT_ADDRESS_ERROR hides the resolution suggestion (see screen shots above) because the "domain" has invalid format, so there's got to be a way to suppress it otherwise the user will not see the address resolved by the SNAP.

As to stating a format of resolution SNAP supports would be great, maybe via regex. We were thinking of doing that check in the SNAP anyway as to not run RPC calls for names that do not fit the format.

**Reply by hmalik88** [reply]
Yes, we will fix that. This is priority for us in our coming sprint, so I'll ping you once we've merged changes 🙂 -- thanks for your input!
